data-prepper-plugins/otel-trace-source/README.md

```diff
@@ -88,8 +88,18 @@ curl -k -H 'Content-Type: application/json; charset=utf-8'  -d '{"resourceSpans"
 ## Metrics
 
 ### Counter
-- `requestTimeouts`: measures total number of requests that time out.
 - `requestsReceived`: measures total number of requests received by otel trace source.
+- `successRequests`: measures total number of requests successfully processed by otel trace source plugin.
+- `badRequests`: measures total number of requests with invalid format processed by otel trace source plugin.
+- `requestTimeouts`: measures total number of requests that time out.
+- `requestsTooLarge`: measures total number of requests of which the number of spans in the content is larger than the buffer capacity.
+- `internalServerError`: measures total number of requests processed by otel trace source with custom exception type.
+
+### Timer
+- `requestProcessDuration`: measures latency of requests processed by otel trace source plugin in seconds.
+
+### Distribution Summary
+- `payloadSize`: measures the distribution of incoming requests payload sizes in bytes.
```

Comment:
What prompted this change? This seems outside of the scope of this work. Does this accomplish another issue that is open?

Reply:
Agree. This is kind of overreach. The initial intention is to add this metric for better monitoring. But to stick with the scope of this work I can pull it off.

Reply:
I might have misread. Do you mean all those new metrics? Some of them like `successRequests, badRequests` are migrated, transformed from processor:

```
### Counter
- `spanProcessingErrors`: records the number of processing exceptions for invalid spans.
- `resourceSpansProcessingErrors`: records the number of processing exceptions for invalid resource spans.
- `totalProcessingErrors`: records the total number of processing errors for spans and resource spans.
```

since we are now decoding the otel-proto request in source. While others like `requestsTooLarge`, `internalServerError` are due to the business logic shift caused by using Span model as Buffer item and calling `Buffer::writeAll` API. `requestProcessDuration` and `payloadSize` might be fully out of the scope although they are for better monitoring. Let me know what you think and I can pull ones off accordingly.